File: YoungPhysicist/YoungPhysicist.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
#include <array>

static void	youngPhysicist(const std::vector<std::vector<int>>& n) {
	const std::array<std::string, 2>	output = {"NO", "YES"};
	
	if (!n.size()) {
		std::cout << output[0] << std::endl;
		return ;
	}

	long	x = 0, y = 0, z = 0;
	for(int row = 1; row < n.size(); row++) {
		x += n[row][0];
		y += n[row][1];
		z += n[row][2];
	}

	const int key = (x == 0 && y == 0 && z == 0);

	std::cout << output[key] << std::endl;
}

int main() {
	std::vector<std::vector<int>>	n;
	std::string						line;;

	while(std::getline(std::cin, line)) {
		std::stringstream	ss(line);
		std::vector<int>	nums;
		int					x;

		while (ss >> x)
			nums.push_back(x);

		n.push_back(nums);
	}
	
	youngPhysicist(n);
	return (0);
}

File: YoungPhysicist/YoungPhysicist.cs
class YoungPhysicist
{
	private static readonly string[] outputs = {"NO", "YES"};
	public static string Solve(List<long[]>	n)
	{
		if (n == null || n.Count == 0)
			return outputs[0];

		long	x = 0, y = 0, z = 0;
		for (int line = 1; line < n.Count; line++)
		{
			x += n[line][0];
			y += n[line][1];
			z += n[line][2];
		}

		int key = (x == 0 && y == 0 && z == 0) ? 1 : 0;

		return outputs[key];
	}
}

class Program
{
	static void Main()
	{
		List<long[]>	n = new();

		string?	line;
		while ((line = System.Console.ReadLine()) != null)
		{
			string[]	input = line.Split(' ');
			long[]	nums = Array.ConvertAll(input, long.Parse);
			n.Add(nums);
		}

		System.Console.WriteLine(YoungPhysicist.Solve(n));
	}
}
